spec/controllers/subjects_controller_spec.rb

```diff
@@ -0,0 +1,211 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe SubjectsController, type: :request do
+  let(:admin) { create_admin_user }
+  let(:teacher) { create_user }
+  let(:listing_url) { '/subjects' }
+  let(:headers) do
+    {
+      'CONTENT_TYPE' => 'application/json',
+      'ACCEPT' => 'application/json'
+    }
+  end
+
+  # Rspec for admin starts here
+  describe 'when a admin user is logged in' do
+    before do
+      login_with_api(admin)
+      @jwt_token = response.headers['Authorization'].split(' ').last
+      headers['Authorization'] = "Bearer #{@jwt_token}"
+    end
+
+    it 'returns a token' do
+      expect(response.headers['Authorization']).to be_present
+    end
+
+    describe 'GET /subjects as admin' do
```

Comment:
@josephelias-rubyians  Write specs for pagination params, see if the relevant data are seen in the given params page